### DISCUSSION TITLE
After set up wizard calibration, z movements still off (cutting too deep)

### DISCUSSION THREAD

**Discussion by rblissilab** [winder/Universal-G-Code-Sender#2117] [Help]
I am trying to get an old Stepoko CNC that I inherited from another teacher back up running for our makerspace.  I have created a simple gcode using Fusion 360 to test out the machine.  The file has a number of pocket clears at different depths.  When I open the file in UGS, it looks good, but when I run the code, it cuts way to deep 2-4 times as much as what it is supposed to do.
I have tried disconnecting and redoing the set up wizard a number of times to calibrate the z direction, but I still get the same issues and cuts.  Are there any further suggestions on how to fix the z displacement?

When calibrating -1 mm is -4.6 mm of displacement.  My model and cut depths were done in inches.

**Comment by breiler** [comment]
And you are now certain that after calibrating, when jogging the Z axis 1mm will actually move the machine 1mm?

If not, which version of UGS are you running? What version of GRBL are you running (type `$I` in the console).

**Reply by rblissilab** [reply]
The z-axis is still moving incorrectly after the calibration.  I hoped trying the wizard a couple of times would get it to calibrate correctly.  Are there other ways to calibrate the system?

The UGS version is 2.0.13 on a mac laptop.  I just went to the downloads section on this [website](https://github.com/winder/Universal-G-Code-Sender) and went with the newest Mac version.

I believe GRBL 0.9j.20160317:

**Reply by breiler** [reply]
There are plenty of ways to calibrate GRBL, just google `calibrate grbl`
Btw, that is a really old version of GRBL you should probably consider upgrading your controller.

**Reply by rblissilab** [reply]
Ok, I will try recalibrating.  If it does not work, I will try upgrading.

**Comment by chrisla23** [comment]
While that is good advice to upgrade grbl, many things have changed since
pre 1.x grbl. Pins got moved around, defaults changed, the flash will get
wiped, etc. It may leave you needing to rewire.

-Chris

**Reply by rblissilab** [reply]
Thanks for the advice.  I will try recalibrating first before upgrading.
